generate_stablehlo.py

```diff
@@ -0,0 +1,278 @@
+from jax._src.sharding_impls import GSPMDSharding
+from jax import tree_util
+
+import sys
+# MLIR python bindings
+# sys.path.append('/home/hanq/git/llvm/llvm-project/build/tools/mlir/python_packages/mlir_core')
+
+from jax._src.lib.mlir import ir
+from jax._src.lib.mlir.dialects import stablehlo
+
+def convert_mlir(bytecode, sample_inputs, sample_outputs):
+    output_tf = tf.convert_to_tensor(sample_outputs)
+    Tout = (output_tf.dtype, )
+    Sout = (tuple(output_tf.shape), )
+    
+    def return_func(*args):
+        return tfxla.call_module(
+            args,
+            version=5,
+            Tout=Tout,  # dtype information
+            Sout=Sout,  # Shape information
+            function_list=[],
+            platforms=('CPU', ),
+            module=bytecode,
+        )
+    return return_func
+
+def fix_mlir_tuple(mlir_module):
+    print(mlir_module.body.operations[0].type)
+    type_ = mlir_module.body.operations[0].type
+    for inst in mlir_module.body.operations[0].body.blocks[0].operations:
+        print(inst)
+
+    if isinstance(type_.results[0], ir.TupleType):
+        with mlir_module.context:
+            mlir_module.body.operations[0].attributes["function_type"] = ir.FunctionType.get(
+                type_.inputs, [type_.results[0].get_type(0)]
+            )
+    
+    mlir_module.dump()
+    return mlir_module
+
+tuple_regex = re.compile(r'tuple<(.*)>')
+xla_shape = re.compile(r'{xla_shape = .*}')
+def fix_hlo_text(hlotext):
+    ## HACK
+    res = []
+    for line in hlotext.split('\n'):
+        if m := tuple_regex.findall(line):
+            line = tuple_regex.sub(m[0], line)
+        line = line.replace('stablehlo.tuple', 'stablehlo.optimization_barrier')
+        line = xla_shape.sub('', line)
+        res.append(line)
+    return '\n'.join(res)
+
+
+class SHLOModel:
+
+    def __init__(self, model_bytecode, pos_to_orig_pos, pos_to_param, sample_output):
+        self.model_bytecode = model_bytecode
+        self.pos_to_orig_pos = pos_to_orig_pos
+        self.pos_to_param = pos_to_param
+        self._total_number = len(pos_to_orig_pos) + len(pos_to_param)
+        output_tf = tf.convert_to_tensor(sample_output)
+        self.tout = (output_tf.dtype, )
+        self.sout = (tuple(output_tf.shape), )
+
+    def __call__(self, *args):
+
+        call_args = []
+        for i in range(self._total_number):
+            if i in self.pos_to_orig_pos:
+                call_args.append(args[self.pos_to_orig_pos[i]])
+            else:
+                call_args.append(self.pos_to_param[i])
+
+        return tfxla.call_module(
+            tuple(call_args),
+            version=5,
+            Tout=self.tout,  # dtype information
+            Sout=self.sout,  # Shape information
+            function_list=[],
+            platforms=('CPU', ),
+            module=self.model_bytecode,
+        )
+        
+        
+
+
+
+def export_torch_model(model: torch.nn.Module, sample_inputs: Tuple):
```

Comment:
I'd add comments to explain each step of the poc for clarity when we share with partners.